[bug] panic: runtime error: invalid memory address or nil pointer dereference

**Describe the bug**

```text
panic: runtime error: invalid memory address or nil pointer dereference                                                                                                                                            
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x6089cb]

goroutine 1 [running]:
main.outCLI.func1(0xc000012480)
        /home/ian/src/todos/internal/cmd/todos/app.go:198 +0xab
github.com/ianlewis/todos/internal/walker.(*TODOWalker).scanFile(0xc0008b7af0, 0xc000040420)
        /home/ian/src/todos/internal/walker/walker.go:282 +0x150
github.com/ianlewis/todos/internal/walker.(*TODOWalker).processFile(0xc0008b7af0, {0xc0010056c0, 0x1a}, {0xc00102c240, 0x28}, 0x2?)
        /home/ian/src/todos/internal/walker/walker.go:263 +0x11c
github.com/ianlewis/todos/internal/walker.(*TODOWalker).walkFunc(0xc0008b7af0, {0xc0010056c0, 0x1a}, {0x6f62b8, 0xc0000a6400}, {0x0?, 0x0?})
        /home/ian/src/todos/internal/walker/walker.go:197 +0x38b
io/fs.walkDir({0x6f4ca8, 0xc0003b20f0}, {0xc0010056c0, 0x1a}, {0x6f62b8, 0xc0000a6400}, 0xc0008b7a00)
        /home/ian/opt/go/src/io/fs/walk.go:73 +0x6c
io/fs.walkDir({0x6f4ca8, 0xc0003b20f0}, {0xc000405aa0, 0x11}, {0x6f62b8, 0xc001206240}, 0xc0008b7a00)
        /home/ian/opt/go/src/io/fs/walk.go:95 +0x2c5 
io/fs.walkDir({0x6f4ca8, 0xc0003b20f0}, {0xc000afc0c0, 0xb}, {0x6f62b8, 0xc0000a7540}, 0xc0008b7a00)
        /home/ian/opt/go/src/io/fs/walk.go:95 +0x2c5 
io/fs.walkDir({0x6f4ca8, 0xc0003b20f0}, {0xc0012c0c60, 0x5}, {0x6f62b8, 0xc000066d00}, 0xc0008b7a00)
        /home/ian/opt/go/src/io/fs/walk.go:95 +0x2c5 
io/fs.walkDir({0x6f4ca8, 0xc0003b20f0}, {0x6f2150, 0x1}, {0x6f63d0, 0xc0003b2100}, 0xc0008b7a00)
        /home/ian/opt/go/src/io/fs/walk.go:95 +0x2c5 
io/fs.WalkDir({0x6f4ca8, 0xc0003b20f0}, {0x6f2150, 0x1}, 0xc000507a00)
        /home/ian/opt/go/src/io/fs/walk.go:122 +0xd2 
github.com/ianlewis/todos/internal/walker.(*TODOWalker).walkDir(...)
        /home/ian/src/todos/internal/walker/walker.go:148
github.com/ianlewis/todos/internal/walker.(*TODOWalker).Walk(0xc0008b7af0)
        /home/ian/src/todos/internal/walker/walker.go:131 +0x1a5
main.newTODOsApp.func1(0xc0000a7940)
        /home/ian/src/todos/internal/cmd/todos/app.go:160 +0x16c
github.com/urfave/cli/v2.(*Command).Run(0xc000205080, 0xc0000a7940, {0xc000016060, 0x2, 0x2})
        /home/ian/src/todos/vendor/github.com/urfave/cli/v2/command.go:274 +0x998
github.com/urfave/cli/v2.(*App).RunContext(0xc0003b0000, {0x6f6248?, 0xa821e0}, {0xc000016060, 0x2, 0x2}) 
        /home/ian/src/todos/vendor/github.com/urfave/cli/v2/app.go:332 +0x5b7
github.com/urfave/cli/v2.(*App).Run(...)
        /home/ian/src/todos/vendor/github.com/urfave/cli/v2/app.go:309
main.main()
        /home/ian/src/todos/internal/cmd/todos/main.go:28 +0x3b
```

**To Reproduce**
Steps to reproduce the behavior:

1. Run `todos` on the kubernetes code base.

**Expected behavior**

`todos` shouldn't crash.

**Additional context**
N/A

--- a/internal/todos/todos.go
+++ b/internal/todos/todos.go
@@ -158,7 +158,7 @@ func (t *TODOScanner) Scan() bool {
 		if next.Multiline {
 			matches := t.findMultilineMatches(next)
 			t.next = append(t.next, matches...)
-			return true
+			return len(t.next) > 0
 		}
 
 		match := t.findLineMatch(next)

--- a/internal/todos/todos_test.go
+++ b/internal/todos/todos_test.go
@@ -508,7 +508,7 @@ func TestTODOScanner(t *testing.T) {
 				},
 			},
 		},
-		"multiline_comments_multiple_todos.go": {
+		"multiline_comments_no_todo.go": {
 			s: &testScanner{
 				comments: []*scanner.Comment{
 					{
@@ -516,7 +516,7 @@ func TestTODOScanner(t *testing.T) {
 						Line: 1,
 					},
 					{
-						Text:      "/*\nfoo\nTODO(github.com/foo/bar/issues1): foo\nTODO: second task\n */",
+						Text:      "/*\nfoo\nfoo\n*/",
 						Line:      5,
 						Multiline: true,
 					},
@@ -529,7 +529,47 @@ func TestTODOScanner(t *testing.T) {
 			config: &Config{
 				Types: []string{"TODO"},
 			},
+			expected: nil,
+		},
+
+		"multiline_comments_multiple_todos.go": {
+			s: &testScanner{
+				comments: []*scanner.Comment{
+					{
+						Text: "// package comment",
+						Line: 1,
+					},
+					{
+						Text: "// TODO: before",
+						Line: 2,
+					},
+					{
+						Text:      "/*\nfoo\nTODO(github.com/foo/bar/issues1): foo\nTODO: second task\n */",
+						Line:      5,
+						Multiline: true,
+					},
+					{
+						Text: "// godoc ",
+						Line: 9,
+					},
+					{
+						Text: "// TODO: after",
+						Line: 10,
+					},
+				},
+			},
+			config: &Config{
+				Types: []string{"TODO"},
+			},
 			expected: []*TODO{
+				{
+					Type:        "TODO",
+					Text:        "// TODO: before",
+					Label:       "",
+					Message:     "before",
+					Line:        2,
+					CommentLine: 2,
+				},
 				{
 					Type:        "TODO",
 					Text:        "TODO(github.com/foo/bar/issues1): foo",
@@ -545,6 +585,14 @@ func TestTODOScanner(t *testing.T) {
 					Message:     "second task",
 					Line:        8,
 					CommentLine: 5,
+				},
+				{
+					Type:        "TODO",
+					Text:        "// TODO: after",
+					Label:       "",
+					Message:     "after",
+					Line:        10,
+					CommentLine: 10,
 				},
 			},
 		},

